loggers/archive.py

```diff
@@ -0,0 +1,70 @@
+from peewee import fn, IntegrityError
+from db import db, archive_db
+from datetime import datetime, timedelta
+import logging
+
+
+def ensure_datetime(timestamp):
+    if isinstance(timestamp, datetime):
+        return timestamp
+    else:
+        return datetime.fromtimestamp(timestamp)
+
+
+@db.atomic()
+def archivable_records_query(Model, cutoff):
+    query = (Model
+             .select()
+             .where(Model.timestamp < cutoff)
+             .order_by(Model.timestamp.asc())
+             .dicts())  # we want a dict for ceating new record
+    return query
+
+
+@archive_db.atomic()
+def archive_record(record_as_dict, ArchiveModel):
+    # ensure Table exists in archive
+    ArchiveModel.create_table(fail_silently=True)
```

Comment:
you definitely don't want to create_table inside the for loop for each record!  an extra query per row :(